### DISCUSSION TITLE
Configure swagger url

### DISCUSSION THREAD

**Discussion by avinashs433** [Burgyn/MMLib.SwaggerForOcelot#171] [Q&A]
Hi,

Is it possible to configure the swagger path to something other than {url}/swagger ?
Maybe to something like {url}/api/v1/swagger ? If so how can I do this ?

Regards,
Avinash

**Comment by Burgyn** [comment]
Hi @avinashs433,

thanks for your question.

Please try the following configuration:

```csharp
app.UseSwaggerForOcelotUI(opt =>
{
    opt.RoutePrefix = "api/v1/swagger";
});
```

Thanks,
Miňo.

**Reply by avinashs433** [reply]
Thanks that works. I can load swagger with
https://localhost:7001/api/v1/swagger/index.html

but how to I get the definitions to load with /api/v1 in it as well ?

It loads with:
https://localhost:7000/swagger/docs/v1/notifications

I want:
https://localhost:7000/api/v1/swagger/docs/v1/notifications

```
"SwaggerEndPoints": [
    {
      "Key": "notifications",
      "Config": [
        {
          "Name": "Notifications API",
          "Version": "v1",
          "Url": "http://notifications.api/swagger/v1/swagger.json"
          
        }
      ]
    }
]

```

```
  app.UseSwaggerForOcelotUI(opt =>
  {
      opt.PathToSwaggerGenerator = "/swagger/docs";
      opt.RoutePrefix = "api/v1/swagger";
  });
```

**Reply by Burgyn** [reply]
Hi @avinashs433,

property `PathToSwaggerGenerator` and `DownstreamSwaggerEndPointBasePath` are the ones you are looking for.

Please try this configuration:

```csharp
app.UseSwaggerForOcelotUI(opt =>
{
    opt.PathToSwaggerGenerator = "/api/v1/swagger/docs";
    opt.DownstreamSwaggerEndPointBasePath = "/api/v1/swagger/docs";
    opt.RoutePrefix = "api/v1/swagger";
});
```

**Reply by GuillermoMarcel** [reply]
I Cannot find any "RoutePrefix" in that object

using this version
"MMLib.SwaggerForOcelot" Version="6.1.0"

![image](https://user-images.githubusercontent.com/10965656/211109635-8c093743-dd22-43f0-9063-8efc1c47c4e8.png)

**Reply by Burgyn** [reply]
Hi @GuillermoMarcel,

from version v6.0.0 you have to use overload of this method with `Action<SwaggerUIOptions> setupUiAction = null` parameter

```csharp
app.UseSwaggerForOcelotUI(opt =>
  {
      opt.DownstreamSwaggerHeaders = new[]
      {
          new KeyValuePair<string, string>("Key", "Value"),
          new KeyValuePair<string, string>("Key2", "Value2"),
      };
  }, c =>
  {
      c.RoutePrefix = "api/v1/swagger";
  });
```

**Reply by vanshsehgal2001** [reply]
Continuing on this, I have a follow up question.
I have configured Ocelot in my existing web app which has controllers of it's own. Thanks to your reply on this where I can Route my ocelot swagger endpoints using this and when I run my project, my project's controllers can render there. Let's say my project's swagger(controllers) render on http://localhost:5000/swagger/index.html while my ocelot's swagger on http://localhost:5000/ocelot-swagger/index.html . Is there a way I can combine both swaggers together and show them together on one swagger page let's say show the ocelot's swagger along with projects controllers on http://localhost:5000/swagger/index.html? What configuration would I need to update to solve this?

**Comment by RiantoFK** [comment]
Hi,

I encounter similar problem:

The Ocelot is available on:
http://localhost:12321/ChipAPI/swagger/index.html

While swagger for service is on:
http://localhost:12321/ChipAPI/swagger/docs/v1/vcta-api

But the library failed to load API definition:
Fetch error
Not Found /swagger/docs/v1/vcta-api

How can I route it to /ChipAPI/swagger/docs/v1/vcta-api instead?

**Reply by Burgyn** [reply]
Hi @RiantoFK,

similar to the previous answer. 
Please try this configuration:

```csharp
app.UseSwaggerForOcelotUI(opt =>
{
    opt.PathToSwaggerGenerator = "/ChipAPI/swagger/docs";
    opt.DownstreamSwaggerEndPointBasePath = "/ChipAPI/swagger/docs";
    opt.RoutePrefix = "ChipAPI/swagger";
});
```

**Reply by GuillermoMarcel** [reply]
I cannot find that property in that object

![image](https://user-images.githubusercontent.com/10965656/211109701-8d48b25f-6e5c-455e-93bf-6eae7986a3b3.png)
